### DISCUSSION TITLE
Elastic Modulus as a variable

### DISCUSSION THREAD

**Discussion by alexchen4ai** [idaholab/moose#18830] [Q&A Modules: Solid mechanics] [answered]
I am using Automatic Differentiation in Moose. 

I used tensor mechanics modules. The elastic modulus are dependent on another variable. It seems that we could only enter the values of C_{ijkl} as float number. Is there a way to input a coupled variable for C_{ijkl}?
```
  [elasticity_tensor]
    type = ADComputeElasticityTensor
    C_ijkl = '10868. 2457. 3279. 21613. 5000.'
    fill_method = axisymmetric_rz
  []
```
  For example, I want to enter something like C_ijkl = '10868.*(1-x) 2457. 3279. 21613. 5000.' where x is another variables.
It doesn't work. Are there some tricks to do this?

**Comment by souravmat-git** [comment]
Is "x" a moose variable or a spatial variable ?

If "x" is declared as a moose variable then check the following material and input file:
moose/modules/tensor_mechanics/src/materials/ComputeConcentrationDependentElasticityTensor.C
moose/modules/combined/test/tests/concentration_dependent_elasticity_tensor/

You will have to modify "_elatictity_tensor" and "_delasticity_tensor_dc" in the material file according to your variable-dependent expression for elasticity tensor.

**Reply by alexchen4ai** [reply]
Thank you! This helps!

**Comment by ngrilli** [comment, marked as answer]
Dear @AlexCHEN-Engineer 

You can check this example in my repository, where I set the elasticity tensor as dependent on the temperature:
https://github.com/ngrilli/c_pfor_am/blob/main/src/materials/ComputeElasticityTensorCPGrain.C

In my case temperature is an AuxVariable, but the same can be done with a MOOSE Variable.

Trust this helps.

Best Regards,
Nicolò Grilli

**Reply by alexchen4ai** [reply]
Thank you for sharing and it makes sense!
